substrate/primitives/weights/src/lib.rs

```diff
@@ -51,23 +51,58 @@ pub mod constants {
 
 /// The weight of database operations that the runtime can invoke.
 ///
-/// NOTE: This is currently only measured in computational time, and will probably
-/// be updated all together once proof size is accounted for.
+/// NOTE: `RuntimeDbRefTime` formerly `RuntimeDbWeight` can only be used to estimate the
+/// computational time component (`ref_time`) of database operations.
+///
+/// It is primarily intended for use within benchmark-generated [`Weight`] functions to help
+/// calculate the computational cost of these operations.
+/// It should NOT be used in any context that is meant to be parachain compatible, where a non-zero
+/// `pov_weight` component is expected.
 #[derive(Clone, Copy, Eq, PartialEq, Default, RuntimeDebug, Encode, Decode, TypeInfo)]
-pub struct RuntimeDbWeight {
+pub struct RuntimeDbRefTime {
```

Comment:
It is more correct like this, but it will break lots of stuff. Can we maybe start by deprecating first?

Reply:
how so, since we export both the deprecated and renamed struct?

Reply:
Maybe it is possible to deprecate the re-export? But it is always better to first deprecate and allow for a slow migration instead of breaking peoples code and forcing them over.

Reply:
Ideally in the first step everyone's code still compiles, and they will just see a warning. 

This PR achieves that as far as I can see here.

Reply:
> Ideally in the first step everyone's code still compiles, and they will just see a warning.
> 
> This PR achieves that as far as I can see here.

Exactly the re-export with the original name is below 
https://github.com/paritytech/polkadot-sdk/blob/dce76a862622c5d8fc3b5a09efbdcab39e20c416/substrate/primitives/weights/src/lib.rs#L67-L68

@ggwpez let me know if you wanted to do that differently